plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcRecordCursor.java

```diff
@@ -141,10 +144,12 @@ public boolean advanceNextPosition()
 
         try {
             if (resultSet == null) {
-                Future<ResultSet> resultSetFuture = executor.submit(() -> {
+                long start = System.nanoTime();
+                ListenableFuture<ResultSet> resultSetFuture = executor.submit(() -> {
                     log.debug("Executing: %s", statement);
                     return statement.executeQuery();
                 });
+                resultSetFuture.addListener(() -> readTimeNanos.addAndGet(System.nanoTime() - start), directExecutor());
```

Comment:
Is it possible to avoid the need for a `ListenableFuture` here by recording the time taken in a `finally` block ?